pkg/controller/stacks/install/stackinstall.go

```diff
@@ -255,24 +255,25 @@ func (h *stackInstallHandler) update(ctx context.Context) (reconcile.Result, err
 // This function ensures that all the resources (e.g., CRDs) that this StackInstall owns
 // are also cleaned up.
 func (h *stackInstallHandler) delete(ctx context.Context) (reconcile.Result, error) {
-	gvk := h.ext.GroupVersionKind()
-	labels := map[string]string{
-		labelParentGroup:     gvk.Group,
-		labelParentVersion:   gvk.Version,
-		labelParentKind:      gvk.Kind,
-		labelParentNamespace: h.ext.GetNamespace(),
-		labelParentName:      h.ext.GetName(),
-		labelParentUID:       string(h.ext.GetUID()),
+	labels := stacks.ParentLabels(h.ext)
+	if err := h.kube.DeleteAllOf(ctx, &v1alpha1.Stack{}, client.InNamespace(h.ext.GetNamespace()), client.MatchingLabels(labels)); runtimeresource.IgnoreNotFound(err) != nil {
```

Comment:
Did you notice how this sets us up to have a single StackInstall resolving its requests/dependencies through multiple Stacks ;-) .. Ofcourse, the `status` of a `StackInstall` doesn't support multiple Stacks (yet).

It is definitely worth commenting that this could potentially remove multiple resources, and that one day that may be the case.